### DISCUSSION TITLE
Typo on page 38?

### DISCUSSION THREAD

**Discussion by artyomboyko** [rasbt/machine-learning-book#80] [General]
Good day Sebastian, can you explain in detail how you got in the note -1 in dL/db? Where does the sum sign (sigma) disappear?

![38](https://user-images.githubusercontent.com/17560478/189044683-90dc6e74-0f45-4c31-8351-58fe9e9b2b8f.png)

**Comment by artyomboyko** [comment]
Yes. There's a typo there. Found the link.
https://github.com/rasbt/machine-learning-book/tree/main/ERRATA

**Reply by rasbt** [reply]
Sorry for the typo, but I am glad you already found it!

**Comment by artyomboyko** [comment]
Can someone explain in detail how it turns out:
$\frac{d}{dw_{j}} \sum_{j} w_{j} x_{j}^{i} = x_{j}^{i}$

How did you get rid of the sigma sign (sum)?

**Reply by rasbt** [reply]
That's a perfectly good question and not obvious at first glance. So the reason why we can eliminate the sum over j is that we are looking for a specific partial derivative wrt to a specific "w_j". So, the other w_j values are basically constants.

Or maybe just think of it like this where 

- a = w_j with j=1
- c = w_j with j=2
- d = w_j with j=3

So 
z = a * x_1 + c * x_3 + d * x_2 + b 

and we look for the partial derivative of  L wrt a. Then the other terms will become constants, so we can cancel c and d and b

**Reply by artyomboyko** [reply]
Good day Sebastian. Thanks for your reply.

I don't have enough experience with this kind of thing. Can I write it down as follows:

Since at the moment of computation we are looking for the partial derivative for a particular example from a dataset, we can exclude $\sum_{j}$. Because all other terms of this sum at the moment of calculation will be constants. Therefore, we can write down the following:

$\frac{d}{dw_{j}} \sum_{j} w_{j} x_{j}^{(i)} = \frac{d}{dw_{j}} w_{j} x_{j}^{(i)} = [Using: \quad(f(x) \cdot g(x))' = f'(x) \cdot g(x) + f(x) \cdot g'(x); \quad (x)' = 1; \quad (C)'= 0 ] = $    

$$ = \frac{d}{dw_{j}} (w_{j}) \cdot  x_{j}^{(i)} + w_{j} \cdot \frac{d}{dw_{j}}(x_{j}^{(i)}) =  1 \cdot x_{j}^{(i)} + w_{j} \cdot 0 = x_{j}^{(i)} $$    



Would it be correct in this manner?

**Reply by rasbt** [reply]
yes this looks correct to me

**Reply by artyomboyko** [reply]
Thank you so much for your help Sebastian.

**Reply by artyomboyko** [reply]
Good day Sebastian. How you get -1 on d/db?

$\frac{d}{db} \sum_{j} w_{j} x_{j}^{(i)} = \frac{d}{db} w_{j} x_{j}^{(i)} = [Using: \quad(f(x) \cdot g(x))' = f'(x) \cdot g(x) + f(x) \cdot g'(x); \quad (x)' = 1; \quad (C)'= 0 ] = $    
 
$= \frac{d}{db} (w_{j}) \cdot  x_{j}^{(i)} + w_{j} \cdot \frac{d}{db}( x_{j}^{(i)} ) = 0 \cdot x_{j}^{(i)} + w_{j} \cdot 0 = 0 + 0$

Then 

$\frac{d}{dw_{j}} (y^{(i)} - \sum_{i} w_j x_{j}^{(i)}+b) = [(x)' = 1, \quad (c)' = 0] = 0 - 0 + 1 = 1$

**Reply by rasbt** [reply]
I think in the last step it should be 0 - (0+1) instead of 0 - 0+1
